------------------------------------------------------------
revno: 14161
revision-id: squid3@treenet.co.nz-20170529053359-xtbuev2zwmdfj9mp
parent: squid3@treenet.co.nz-20170529043852-zkf91gxhaqdj0rkn
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4682
author: Christos Tsantilas <chtsanti@users.sourceforge.net>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Mon 2017-05-29 17:33:59 +1200
message:
  Bug 4653: %st lies about tunneled traffic volumes
  
  Squid-5 and squid-4 does not count the "HTTP/1.1 200 Connection Established"
  header size for %<st formatting code.
  
  This is a Measurement Factory project
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170529053359-xtbuev2zwmdfj9mp
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: c340785d0d5042ae0f783d606f0998d605290ac4
# timestamp: 2017-05-29 05:51:04 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20170529043852-\
#   zkf91gxhaqdj0rkn
# 
# Begin patch
=== modified file 'src/tunnel.cc'
--- src/tunnel.cc	2017-01-01 00:16:45 +0000
+++ src/tunnel.cc	2017-05-29 05:33:59 +0000
@@ -836,7 +836,7 @@
  * Call the tunnelStartShoveling to start the blind pump.
  */
 static void
-tunnelConnectedWriteDone(const Comm::ConnectionPointer &conn, char *buf, size_t size, Comm::Flag flag, int xerrno, void *data)
+tunnelConnectedWriteDone(const Comm::ConnectionPointer &conn, char *, size_t len, Comm::Flag flag, int, void *data)
 {
     TunnelStateData *tunnelState = (TunnelStateData *)data;
     debugs(26, 3, HERE << conn << ", flag=" << flag);
@@ -848,6 +848,11 @@
         return;
     }
 
+    if (ClientHttpRequest *http = tunnelState->http.get()) {
+        http->out.headers_sz += len;
+        http->out.size += len;
+    }
+
     tunnelStartShoveling(tunnelState);
 }
 

